pandas/tests/indexes/common.py

```diff
@@ -205,6 +205,39 @@ def test_hash_error(self):
                                        type(ind).__name__):
                 hash(ind)
 
+    def test_copy(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            if isinstance(index, MultiIndex):
+                continue
+
```

Comment:
OK for PI, not for RI I think (it doesn't have any data to copy), and for MI I have the test ready but at the moment it would fail since this PR doesn't include code for MI. Will include in a separate PR.